### ISSUE TITLE
Interface doesn't work and strange behavior

### ISSUE THREAD

**Issue by exa18** [naev/naev#1706]
version 0.8.2+dev
system: Windows 10

Toggling Info->Weapons secondary does not work, as well as automatilcy assign numeric shortcat for exp. to affterburn.
In battle fire weapons doesn't work (default inputs).

Diplomacy with positive standing value performs attack.

Used save file from upgrade from 0.8.0.

**Comment by UncombedCoconut** [comment]
I'd like to help, but I'm having trouble understanding this report.
It might help if we could see this save file and your keyboard settings. (You can point Windows Explorer at `%APPDATA%\naev` -- then zip up the `conf.lua` file and the relevant file in the `saves` folder.)
What specifically is happening -- something weird when you select weapon-sets 1-9/0 in the list and left/right click the weapons?
Something weird afterward when you try to fire the weapons?

> Diplomacy with positive standing value performs attack.

This is a separate problem, right? Who is attacking you? Does it happen spontaneously or only under certain conditions?

**Comment by exa18** [comment]
Weapons not respond to keyboard. And when I enter Info menu -> weapons to set primary/secondary it can't be set. Right click do not select secondary and binding to 1-9 for accesories too.

Dvaered with standing +2% should be neutral and before update nothing like that happen.
When I takeoff from they planet, few seconds later attacking me "Dvaered Police Vendeta".


[naev.zip](https://github.com/naev/naev/files/6027262/naev.zip)

**Comment by nenau-again** [comment]
Aren't you maybe in the middle of the Baron's holopainting mission ?
That could explain the vendetta attacking you. Not the weapons problem.

**Comment by exa18** [comment]
Yes, true with Baron's mission :).
But I could not respond with fire and setup primary/secondary weapons too.

**Comment by nenau-again** [comment]
What happens when you try to de-check "enable instant fire" ?
Or alternatively, check "automatic handle weapons"
in your weapon set menu.
Because your weapon handling settings are odd.

**Comment by exa18** [comment]
Enabling automatic handle unlocks fire. So I can check secondary weapons set and both keyboard shortcuts for fire work now.
And shortcut for my burner also show.

Ok. I reproduced. It true checked instatnt mode blocks weapons fire. Uncheck this unlocks fire and enables possibility to selecet primary/secondary. 
Problem is option instant mode which disables fire thru shortcats and weapons don't shoot at all (even to targets).

**Comment by onpon4** [comment]
> Problem is option instant mode which disables fire thru shortcats and weapons don't shoot at all (even to targets).

Instant mode causes a weapon to "instantly" fire when you press its associated number key. It's not terribly useful but the fact that a weapon group in instant mode doesn't fire with the space bar is in fact intentional behavior. Set a weapon to group 4 in instant mode, for instance, and pressing the 4 key will fire the weapon.

This is poorly documented though. I think we probably need a tutorial of some kind somewhere in the game explaining how weapons configuration works. (Placement is tricky since the game is designed in such a way that you can play through the game just fine without configuring weapons, just leaving automatic handling on; it's therefore not entirely clear when to introduce this to the player. Perhaps somehow fitting it into the weapons configuration window itself would be nice, though I can't think of a good way to do this at the moment.)

🕵️

**Comment by exa18** [comment]
Help button to clarify it would be nice add.

**Comment by amylaar** [comment]
> This is poorly documented though.
Huh, indeed.  I didn't know this, even after years of on-and-off playing Naev.

> I think we probably need a tutorial of some kind somewhere in the game explaining how weapons configuration works. (Placement is tricky since the game is designed in such a way that you can play through the game just fine without configuring weapons, just leaving automatic handling on; it's therefore not entirely clear when to introduce this to the player. Perhaps somehow fitting it into the weapons configuration window itself would be nice, though I can't think of a good way to do this at the moment. EDIT: Maybe a "Help" button would do it?)

I think this ties in with the problem that the current starting tutorial is way to incomplete overall, and that you can't re-take
it in part or in whole withput starting a new game.
IIRC there was an issue(?) somewhere for that, but it's closed, even though it's a serious regression and an ongoing problem; I commented there that disabling is not covered.
If we had again a tutorial with multiple parts that you can take and re-visit as desired (be that from the main menue or from (a set of) place(s) in the game, e.g. Melendez dealer shipyards), it would be straightforward to add another lesson to the list.

Also, one way to make the thing appear a bit more organic is to require to pass relevant lessons to aquire licenses for
heavy weapons and combat vessels.  Maybe add a new license for switched outfits, make it cheap but require a lesson completed.  Or just give a reminder of the lessons available when buying licenses / outfits.

**Comment by onpon4** [comment]
My thoughts on what makes a good tutorial:

1. It should be comprehensive. The old tutorial met this standard a long time ago, but became less and less comprehensive over time. The new tutorial, of course, has not yet ever met this condition.
2. It should teach things as they become relevant. Otherwise, the player is likely to forget the information shortly after learning it (as we tend to avoid remembering information that we have no use for). This right here is why I thought it necessary to completely replace the old tutorial, and why I still think that was the right decision despite the incompleteness of the new one as of yet: it didn't teach things in context, but instead it gave you a giant menu of things to learn.
3. It should be as unobtrusive and seamless as possible. The tutorial should ideally feel like a natural part of the game. The old tutorial did not meet this standard, but the new one does.
4. The tutorial must be optional or skippable. All tutorials that have ever existed for Naev have met this standard.

Regarding re-taking tutorials, that can be useful at times when a concept is hard to learn or easy to overlook in particular. Another option in such a situation is simple repetition. Shadow the Hedgehog, for instance, reminds you forever about various mechanics as they become relevant via little help bubbles. (I know, not the best game to reference, but it does teach its mechanics well.)

Of course, not every tutorial will need to be repeated. The tutorial on basic flight for example almost certainly only needs to be seen once since flight is something you're guaranteed to do a lot. I think it's worth identifying information that can be easy to get and implementing reminders or ways to re-learn on a targeted basis.

> IIRC there was an issue(?) somewhere for that, but it's closed, even though it's a serious regression and an ongoing problem; I commented there that disabling is not covered.

I couldn't find it. But yes, how to disable ships isn't covered yet. I think this should be covered after the player has already played for awhile, perhaps with a mission that requires you to disable something. The Teenager mission strikes me as a possible mission that can do this, though since that one's specific to Dvaered space, it might make sense to create an entirely new mission for it.

> If we had again a tutorial with multiple parts that you can take and re-visit as desired (be that from the main menue or from (a set of) place(s) in the game, e.g. Melendez dealer shipyards), it would be straightforward to add another lesson to the list.

I have a couple problems with that. Firstly, if it's available to new players, it can be overwhelming to see all this out-of-context information in a giant menu. That can be worked around easily enough by making it a Lua menu in-game. But the other problem I have with it is it feels to me like compensation for the game not teaching mechanics properly in the first place. I think repetition should be achieved by allowing the player to get multiple overviews from multiple in-context places rather than a player being expected to redo the exact same tutorial again, and if a tutorial doesn't really need repetition, a player finding that they need a refresher is likely a sign that the concept wasn't taught in an appropriate context.

> Also, one way to make the thing appear a bit more organic is to require to pass relevant lessons to aquire licenses for
heavy weapons and combat vessels. Maybe add a new license for switched outfits, make it cheap but require a lesson completed. Or just give a reminder of the lessons available when buying licenses / outfits.

An interesting idea for sure. A major problem, though, is it wouldn't allow you to put your knowledge into practice within these tutorials. It would effectively be like having to get a driver's license before you're allowed into the driver's seat. There are tricks that could be used to work around this, but they would be clunky I think (like giving the player a special ship that has the outfits or something). This would also possibly feel like a mandatory tutorial, which I think needs to be avoided.

🕵️

**Comment by amylaar** [comment]
>     2. It should teach things as they become relevant. Otherwise, the player is likely to forget the information shortly after learning it (as we tend to avoid remembering information that we have no use for). This right here is why I thought it necessary to completely replace the old tutorial, and why I still think that was the right decision despite the incompleteness of the new one as of yet: it didn't teach things in context, but instead it gave you a giant menu of things to learn.

I suppose you could tag missions with a set of relevant lessons, and make these available at the time a mission is offered.
A successfully completed mission could also set an 'equivalent experience' flag for the lesson content, so you can give reminders
when a mission is offered / taken if neither the lesson nor a mission that is considered equivalent experience has been
completed.
> 
>     3. It should be as unobtrusive and seamless as possible. The tutorial should ideally feel like a natural part of the game. The old tutorial did not meet this standard, but the new one does.

You might get to the lessons menue from the mission computer and/or the outfitter.
> 
>     4. The tutorial must be optional or skippable. All tutorials that have ever existed for Naev have met this standard.
<pedantic> Actually, there was a set of Naev versions where the default key binding was empty, so if you didn't
explicitly set a keybinding before, you were stuck inside the tutorial with no way out but to kill Naev (if you could still access other windows) or reboot.  </pedantic>
 
> Of course, not every tutorial will need to be repeated. The tutorial on basic flight for example almost certainly only needs to be seen once since flight is something you're guaranteed to do a lot. I think it's worth identifying information that can be easy to forget and implementing reminders or ways to re-learn on a targeted basis.

That is only true if you continually live and breath Naev.  If you pursue other interests for some years and then come back to
your old saved game, you can be completely out of practice, so it's nice if you can make an appointment with Captain T. Practice
to jog your memory.
> 
> > IIRC there was an issue(?) somewhere for that, but it's closed, even though it's a serious regression and an ongoing problem; I commented there that disabling is not covered.
> 
> I couldn't find it. But yes, how to disable ships isn't covered yet. I think this should be covered after the player has already played for awhile, perhaps with a mission that requires you to disable something. The Teenager mission strikes me as a possible mission that can do this, though since that one's specific to Dvaered space, it might make sense to create an entirely new mission for it.
> 
> > If we had again a tutorial with multiple parts that you can take and re-visit as desired (be that from the main menue or from (a set of) place(s) in the game, e.g. Melendez dealer shipyards), it would be straightforward to add another lesson to the list.
> 
> I have a couple problems with that. Firstly, if it's available to new players, it can be overwhelming to see all this out-of-context information in a giant menu. That can be worked around easily enough by making it a Lua menu in-game. But the other problem I have with it is it feels to me like compensation for the game not teaching mechanics properly in the first place. I think repetition should be achieved by allowing the player to get multiple overviews from multiple in-context places rather than a player being expected to redo the exact same tutorial again, and if a tutorial doesn't really need repetition, a player finding that they need a refresher is likely a sign that the concept wasn't taught in an appropriate context.
> 
> > Also, one way to make the thing appear a bit more organic is to require to pass relevant lessons to aquire licenses for
> > heavy weapons and combat vessels. Maybe add a new license for switched outfits, make it cheap but require a lesson completed. Or just give a reminder of the lessons available when buying licenses / outfits.
> 
> An interesting idea for sure. A major problem, though, is it wouldn't allow you to put your knowledge into practice within these tutorials.
A good point. If a tutorial requires special equipment, the lesson should only become available when you aquire the necessary
equipment.  That also gives a natural point where the lessons menue is expanded.
This opens another problem, though: You might be missing important information what equipment to buy to complete a mission.
The hint that you need ion weapons for disabling was in the old disable lesson.  If the lesson becomes contingent on buying
an ion weapon or one of the new modifier outfits, there need to be hints on the need for these elsewhere.
To stay with the teenager mission, once you accept the mission, there could appear another NPC in the bar that gives you
a hint about the needed equipment if you don't have it already.  It might also offer to sell you some pieces of equipment if it's
not available regularly from the local outfitters.
Actually, that's also a bit of a problem with the Dvaered championship - you need ion cannons, or even better a Lancelot with a turreted ion cannon, to win this, but it's not locally available there, so you can get trapped to loose there.  An NPC weapons dealer to fill the gap would be quite helpful there.

**Comment by UncombedCoconut** [comment]
> That is only true if you continually live and breathe Naev. If you pursue other interests for some years and then come back to
> your old saved game, you can be completely out of practice, so it's nice if you can make an appointment with Captain T. Practice
> to jog your memory.

Several good points/ideas in this thread.
Regarding this specific one: no problem, start a new game if you want to re-experience the start of the game.

**Comment by amylaar** [comment]
> The hint that you need ion weapons for disabling was in the old disable lesson. If the lesson becomes contingent on buying
> an ion weapon or one of the new modifier outfits, there need to be hints on the need for these elsewhere.

I just came up with another idea to give such a hint: At the outfitters, in the equipment list, for any key equipment, you could state that it comes with access to a *free* tutorial lesson for <insert lesson name>.  Make the icon for the outfit look like a special offer promotional packaging if you haven't unlocked the lesson yet.

**Comment by bobbens** [comment]
See as this is more of a need better explaining mechanics rather than an interface bug, I created a new issue #1911 which hopefully gathers most of the ideas here and will use as a base for improving tutorial / documentation. Closing this issue, thanks!
